docs/ocean/design_docs/vert-lagrang-remap/vert-lagrang-remap.rst

```diff
@@ -0,0 +1,524 @@
+   
+   vertVelocityTop = (layerThickness(tlev=2) - layerThicknessALE)/dt
+
+The computation of :code:`vertTransportVelocityTop` and 
+:code:`vertGMBolusVelocitytop` is unchanged as these fields represent Eulerian 
+velocities.
+
+Implementation: Grid is sufficiently conditioned for accuracy and stability
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/15
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+After determining the target grid, perform the following steps:
+
+#. Optional: Assign :math:`h_k^{t+1}` to :math:`h_k^{lg}` if 
+   :math:`h_k^{t+1} - h_k^{lg}` is less than a minimum thickness alteration. 
+   This motivated by accuracy considerations, as each remapping may introduce 
+   errors. *Darren, would this improve PPR computational performance?*
+#. Apply minimum layer thickness criterion. 
+
+Smoothing layers in space and time is left for a future design document in 
+which we implement support for additional coordinate systems including hybrid 
+coordinates.
+
+Namelist options:
+
+- Minimum layer thickness
+- Optional: minimum thickness change for remapping to occur
+
+
+Implementation: Ability to specify remapping method and its options
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/15
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+Namelist options:
+
+- frequency with which remapping should be performed (on which timestep)
+- order of the remapping
+- order of edge slope estimates
+- monotone limiter
+- boundary condition option
+- option to output some diagnostics?
+- *Some other remapping options here*
+
+
+Implementation: Support for existing time stepping schemes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/15
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+:code:`vertAleTransportTop` is set to zero for both time stepping schemes from 
+:code:`ocn_vert_transport_velocity_top`.
+
+*Some details here about how to treat ALE_thickness*
+
+Implementation: performance
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/15
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+Options for improving performance:
+
+- Using the split-explicit scheme
+- Splitting the scalar and momentum timesteps
+- Only remapping when the change in thickness exceeds given threshold
+- Optimizing/parallelizing PPR?
+
+Implementation: no interference
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/15
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+Ensure that no calculations are made outside of remapping itself using 
+prognostic (or diagnostic) variables both before and after remapping to avoid 
+introducing additional errors.
+
+Look for places in the code where prognostic variables are used at the previous 
+timestep.
```

Comment:
One way of detecting these would be to run a test where you set all the prognostic variables at the previous time level to garbage (`-1e34` is a popular choice in MPAS-Ocean) and make sure nothing breaks.

Reply:
Yes, I agree. I wrote something along these lines in the testing section.

Reply:
Ah, sorry, I didn't get that far.